### ISSUE TITLE
HTTP_X_FORWARDED_PROTO header is not taken into account

### ISSUE THREAD

**Issue by beda42** [palcilibraries/CC-Plus#4]
We run CC-Plus behind a reverse proxy which is used for HTTPS and the CC-Plus server itself serves over unencrypted HTTP only. Unfortunately, it is not possible to run it properly in such a setup without extra configuration, because CC-Plus does not take the HTTP_X_FORWARDED_PROTO header into account when creating URLs. It thus creates all the URLs with the "http://" protocol which then breaks for example the login process.

We managed to fix it by enforcing HTTPS signalling on the Apache level (by using `SetEnv HTTPS on`), but this is a sub-optimal solution. Ideally CC-Plus should honor the HTTP_X_FORWARDED_PROTO header if present.

**Comment by lsr2912** [comment]
Just pushed in a change to the middleware, but I don't have a way to test it. It appears to be Laravel-related, and I set things up as-per the [Laravel documentation](https://laravel.com/docs/8.x/requests#configuring-trusted-proxies). Can you test this further?

**Comment by beda42** [comment]
I tried it, but no luck. Then I had a look at the Laravel documentation and found that it required extra configuration that I had not done. After I added my proxy to the list of trusted proxies, it started working. In fact, I think that it would have worked with the previous version as well, so I apologize for the false bug report.

For documentation purposes, this is how I made it work:

```
php artisan vendor:publish --provider="Fideloper\Proxy\TrustedProxyServiceProvider"
```
This created a config file `config/trustedproxy.php` where I edited the `proxies` value to match my proxy IP address and it started working.
